### PR TITLE
feat(interactions): Add confession deleting

### LINKED ISSUE #66: Ability for users to delete their own post

### PR DESCRIPTION
**Pull request by cfanoulis**
Fixes #66

### CODE REVIEW

**Comment by cfanoulis** [comment]
kay vercel, calm down

**Comment by cfanoulis** [comment]
updating this tomorrow to be CRT-only

**Comment by benaubin** [comment]
Awesome. To summarize our discussion in Slack, I believe that only CRT or staff should be able to delete confessions, not OP. A lot of the time, confessions have benefits beyond what they provide to OP, and because CRT took responsibility for posting, I believe it should be up to CRT to determine deletion.

**Comment by cfanoulis** [comment]
note to self: also need to update manifest.yml to show the shortcut

**Comment by cfanoulis** [comment]
As discussed in #confession-meta, ani's official stance on this is keeping this OP only. As such, the PR is once again open for reviews

File: pages/api/interaction_work.ts [hunk 1]
```diff
@@ -301,6 +304,33 @@ export default async function handler(
...
+            data.response_url,
+            "You are not the original poster of the confession, so you cannot react anonymously."
+          );
+          res.writeHead(200).end();
+          return;
+        }
+        await web.chat.delete({
+          channel: confessions_channel,
+          ts: data.message.thread_ts!,
+        });
```
Comment (anirudhb) [inline pages/api/interaction_work.ts:329]: No error checking

File: pages/api/interaction_work.ts [hunk 2]
```diff
@@ -301,6 +304,33 @@ export default async function handler(
...
+        await repo.delete(record);
```
Comment (anirudhb) [inline pages/api/interaction_work.ts:330]: No error checking

**Review by anirudhb — APPROVED** [review]
lgtm, need to test sometime in the next few days

**Pull request closed without merging**